From 2917caa4550ffd8ec9033db2140defd06a868267 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Thu, 19 Apr 2018 14:00:21 -0700
Subject: [PATCH 0104/1851] sound: xilinx: pcm: Set the shorter device name

commit a36e6a683fa12778baae84bd335b352f83bb0b72 from
https://github.com/Xilinx/linux-xlnx.git

By default, the device names derive from the dt description. The DP
pcm devices are represented as a child node of DP node, which results
in both parent and child node names in the device name. The name
created in this way gets too long to fit into the sound component name,
and last characters where IDs are located are not included. This gives
the same component name for two different components, and the debugfs
fails to create entries with same name as warning below:

xilinx-dp-snd-pcm fd4a0000.zynqmp-display:zynqmp_dp_snd_pcm1: ASoC: Failed to create component debugfs directory

This fixes it by setting the child node name as a device name.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Satish Kumar Nagireddy <satishna@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xilinx-dp-pcm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/xilinx/xilinx-dp-pcm.c b/sound/soc/xilinx/xilinx-dp-pcm.c
index 4c2cdfb9d613..518c1d08aca7 100644
--- a/sound/soc/xilinx/xilinx-dp-pcm.c
+++ b/sound/soc/xilinx/xilinx-dp-pcm.c
@@ -46,6 +46,7 @@ static int xilinx_dp_pcm_probe(struct platform_device *pdev)
 {
 	int ret;
 
+	dev_set_name(&pdev->dev, pdev->dev.of_node->name);
 	ret = devm_snd_dmaengine_pcm_register(&pdev->dev,
 					      &xilinx_dmaengine_pcm_config, 0);
 	if (ret)
-- 
2.31.1

